src/core/session.c

```diff
@@ -242,12 +242,12 @@ MsQuicSessionOpen(
     }
 #endif
 
-    if (Session->Registration->AppName[0] != '\0') {
-        char SpecificAppKey[256] = QUIC_SETTING_APP_KEY;
-        strcpy_s(
+    if (Session->Registration->AppNameLength != 0) {
+        char SpecificAppKey[261] = QUIC_SETTING_APP_KEY;

```

Comment:
Make this more readable and future proof by using constants, e.g. `MAX_APP_NAME + sizeof(QUIC_SETTING_APP_KEY)`